File: pinyougou-web/pinyougou-user-web/src/main/webapp/js/controller/userController.js
/** 定义控制器层 */
app.controller('userController', function ($scope, baseService) {
    /** 定义用户对象 */
    $scope.user = {};
    /** 定义保存用户方法 */
    $scope.save = function () {
        if ($scope.rePassword != $scope.user.password) {
            alert("密码不一致，请重新输入！");
            return;
        }
        baseService.sendPost("/user/save?smsCode=" + $scope.smsCode, $scope.user).then(function (response) {
            if (response.data) {
                alert("注册成功！")
                //清空数据
                $scope.user = {};
                $scope.rePassword = "";
                $scope.smsCode = ";"
            } else {
                alert("注册失败！")
            }
        });
    };

    /** 定义发送短信验证码方法 */
    $scope.sendCode = function () {
        if ($scope.user.phone && /^1[3|4|5|6|8]\d{9}$/.test) {
            //发送异步请求
            baseService.sendPost("/user/sendCode?phone=" + $scope.user.phone).then(function (response) {
                if(response.data){
                    alert("发送成功！")
                }else {
                    alert("发送失败！")
                }

            });
        } else {
            alert("手机号码无效！！")
        }
    };

})
;

File: pinyougou-web/pinyougou-user-web/src/main/java/com/pinyougou/user/controller/UserController.java
package com.pinyougou.user.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.User;
import com.pinyougou.service.SmsService;
import com.pinyougou.service.UserService;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * @author NTP
 * @date 2018/12/8
 */
@RestController
@RequestMapping("/user")
public class UserController {

    @Reference(timeout = 10000)
    private UserService userService;

    /** 注册用户 */
    @PostMapping("/save")
    public boolean save(@RequestBody User user,String smsCode) {
        try {
            //判断验证码是否正确
            if (userService.checkSmsCode(user.getPhone(),smsCode)){
                userService.save(user);
                return true;
            }
        }catch (Exception ex){
            ex.printStackTrace();
        }
        return false;
    }

    /** 发送验证码 */
    @PostMapping("/sendCode")
    public boolean sendCode(String phone){
        try {
            /** 发送验证码 */
            return userService.sendCode(phone);
        }catch (Exception ex){
            ex.printStackTrace();
        }
        return false;
    }

}
